src/core/project/qgsproject.cpp

```diff
@@ -82,6 +82,7 @@
 #include <QStandardPaths>
 #include <QUuid>
 #include <QRegularExpression>
+#include <QProgressDialog>
```

Comment:
QProgressDialog is a gui class - it can't be used from the core library. Rather, this kind of widget behaviour belongs up at the app level, responding to signals and progress reports emitted by QgsProject

Reply:
Oh, I was disoriented and assumed QgsProject to be a gui class. Will move QProgressDialog to QgisApp.